packages/contracts/src/contracts/current/forwarder/MixinERC721.sol

```diff
@@ -0,0 +1,33 @@
+pragma solidity ^0.4.24;
+pragma experimental ABIEncoderV2;
+
+import "zeppelin-solidity/contracts/token/ERC721/ERC721Basic.sol";
+import "../protocol/AssetProxy/ERC721Proxy.sol";
+
+contract MixinERC721 is
+    ERC721Proxy
+{
+    // Equals to `bytes4(keccak256("onERC721Received(address,uint256,bytes)"))`
+    bytes4 constant ERC721_RECEIVED = 0xf0b9e5ba;
+
+    function onERC721Received(address, uint256, bytes memory)
+        public
+        pure
+        returns(bytes4)
+    {
+        return ERC721_RECEIVED;
+    }
+
+    function transferERC721Token(
+        bytes memory assetData,
+        address from,
+        address to,
+        uint256 amount
+    )
+        internal
+    {
+        // Pop off the proxy id as this needs to be done prior
+        popLastByte(assetData);
+        ERC721Proxy.transferFromInternal(assetData, from, to, amount);
```

Comment:
I found this confusing, since we're not actually calling the proxy here. Maybe `import { ERC721Proxy as ERC721Transfer} from ...` would make this more clear.

Reply:
This would be solved with #659 which separates the responsibilities of ERC721 Transfer and the AssetProxy. In the meantime I'll import it and alias it.